### ISSUE TITLE
Unable to compile project against rebar 2.5.1

### ISSUE THREAD

**Issue by aforward** [wooga/eredis#73]
I am coming from Elixir, so the stack trace from Erlang is a little foreign to me; sorry if this is a easy fix.

I am able to compile against the provided version of rebar

```bash
root@33175fa19b42:/src/eredis# ./rebar --version
rebar version: 2 date: 20110304_023706 vcs: git d725e5f
root@33175fa19b42:/src/eredis# ./rebar compile
==> eredis (compile)
```

But against the latest version, I am not.

```bash
root@33175fa19b42:/src/eredis# rebar --version
rebar 2.5.1 17 20150421_125738 git 2.5.1-180-g858fb4f
root@33175fa19b42:/src/eredis# rebar compile
==> eredis (compile)
ERROR: compile failed while processing /src/eredis: {'EXIT',
    {function_clause,
        [{filename,join,[[]],[{file,"filename.erl"},{line,392}]},
         {rebar_erlc_compiler,expand_include_lib_path,1,
             [{file,"src/rebar_erlc_compiler.erl"},{line,647}]},
         {rebar_erlc_compiler,process_attr,3,
             [{file,"src/rebar_erlc_compiler.erl"},{line,597}]},
         {rebar_erlc_compiler,parse_attrs,2,
             [{file,"src/rebar_erlc_compiler.erl"},{line,565}]},
         {rebar_erlc_compiler,modify_erlcinfo,4,
             [{file,"src/rebar_erlc_compiler.erl"},{line,400}]},
         {rebar_erlc_compiler,'-update_erlcinfo_fun/2-fun-0-',4,
             [{file,"src/rebar_erlc_compiler.erl"},{line,359}]},
         {lists,foldl,3,[{file,"lists.erl"},{line,1261}]},
         {rebar_erlc_compiler,init_erlcinfo,2,
             [{file,"src/rebar_erlc_compiler.erl"},{line,349}]}]}}
```

**Comment by knutin** [comment]
It looks like a crash inside rebar caused by joining empty filepaths (thats why filename:join/1 crashes).  I'm on vacation for another week, can have a look when I have a computer again :-)

**Comment by aforward** [comment]
Enjoy.

I have downgraded my rebar so I have a solution in place; and thanks for
prompt-even-while-on-holiday response.

andrew f

On Sun, May 17, 2015 at 10:59 PM, Knut Nesheim <notifications@github.com>
wrote:

> It looks like a crash inside rebar caused by joining empty filepaths
> (thats why filename:join/1 crashes). I'm on vacation for another week, can
> have a look when I have a computer again :-)
>
>
>
> > On 17.05.2015, at 07:22, Andrew Forward <notifications@github.com>
> wrote:
> >
> > I am coming from Elixir, so the stack trace from Erlang is a little
> foreign to me; sorry if this is a easy fix.
> >
> > I am able to compile against the provided version of rebar
> >
> > root@33175fa19b42:/src/eredis# ./rebar --version
> > rebar version: 2 date: 20110304_023706 vcs: git d725e5f
> > root@33175fa19b42:/src/eredis# ./rebar compile
> > ==> eredis (compile)
> > But against the latest version, I am not.
> >
> > root@33175fa19b42:/src/eredis# rebar --version
> > rebar 2.5.1 17 20150421_125738 git 2.5.1-180-g858fb4f
> > root@33175fa19b42:/src/eredis# rebar compile
> > ==> eredis (compile)
> > ERROR: compile failed while processing /src/eredis: {'EXIT',
> > {function_clause,
> > [{filename,join,[[]],[{file,"filename.erl"},{line,392}]},
> > {rebar_erlc_compiler,expand_include_lib_path,1,
> > [{file,"src/rebar_erlc_compiler.erl"},{line,647}]},
> > {rebar_erlc_compiler,process_attr,3,
> > [{file,"src/rebar_erlc_compiler.erl"},{line,597}]},
> > {rebar_erlc_compiler,parse_attrs,2,
> > [{file,"src/rebar_erlc_compiler.erl"},{line,565}]},
> > {rebar_erlc_compiler,modify_erlcinfo,4,
> > [{file,"src/rebar_erlc_compiler.erl"},{line,400}]},
> > {rebar_erlc_compiler,'-update_erlcinfo_fun/2-fun-0-',4,
> > [{file,"src/rebar_erlc_compiler.erl"},{line,359}]},
> > {lists,foldl,3,[{file,"lists.erl"},{line,1261}]},
> > {rebar_erlc_compiler,init_erlcinfo,2,
> > [{file,"src/rebar_erlc_compiler.erl"},{line,349}]}]}}
> > —
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wooga/eredis/issues/73#issuecomment-102902743>.
>



-- 
aforward

**Comment by StevenJL** [comment]
I also have this same issue.

**Comment by flyinghail** [comment]
Replace
```
include_lib("eredis.hrl").
```
to
```
include("eredis.hrl").
```
in src/eredis_parser.erl

**Comment by knutin** [comment]
@flyinghail is correct. I just merged #73 which fixes the issue and tagged v1.0.8.
